kong/plugins/redirect/schema.lua

```diff
@@ -0,0 +1,39 @@
+local typedefs = require "kong.db.schema.typedefs"
+
+return {
+    name = "redirect",
+    fields = {
+        {
+            protocols = typedefs.protocols_http
+        },
+        {
+            config = {
+                type = "record",
+                fields = {
+                    {
+                        status_code = {
+                            description = "The response code to send. Must be an integer between 100 and 599.",
+                            type = "integer",
+                            required = true,
+                            default = 301,
+                            between = { 100, 599 }
+                        }
+                    },
+                    {
+                        location = typedefs.url {
+                            description = "The URL to redirect to",
+                            required = true
+                        }
+                    },
+                    {
+                        keep_incoming_path = {
```

Comment:
--Green Hat Idea-- 
What do you think about switching this from `keep_incoming_path` being a boolean to an enum like: `incoming_path_strategy`: `One.of({'preserve', 'combine', 'ignore'})` or something similar? At the beginning we could start with `preserve` and `ignore` (`ignore` being the default).

I'm trying to look a bit ahead and see if a boolean won't limit us to much. At the same time it might introduce flexibility where we don't need it since combining incoming path with the one defined above would require us to answer questions about precedence etc.

Reply:
@nowNick I initially had `combine` as an option, but removed it due to complexity + lack of use case.

I think I'd recommend additional parameters in the future anyway as `combine` can be one of multiple states:

* merge paths (location first)
* merge paths (request first)
* merge query (keep all)
* merge query (prefer location)
* merge query (prefer request)

So we'd end up with `keep_incoming_path: true` and `path_strategy: location_first` + `query_strategy: prefer_location`

`path_strategy` and `query_strategy` would only be valid with `keep_incoming_path: true`